content/departments/product-engineering/engineering/cloud/delivery/managed/resume_process.md

```diff
@@ -0,0 +1,114 @@
+  - [3) Verify instance heath](#3-verify-instance-heath)
+  - [4) Wrapping up](#4-wrapping-up)
+
+## General setup
+
+Managed instances configuration is tracked in [deploy-sourcegraph-managed] - make sure you have the latest revision of this repository checked out. For basic operations like accessing an instance for these steps, see [managed instances operations](operations.md).
+
+First, ensure you have the prerequisites installed and up-to-date:
+
+- [Terraform CLI](https://www.terraform.io/)
+- [Deno CLI](https://deno.land/) (used for scripting upgrades)
+- [Comby CLI](https://comby.dev/) (used for rewriting configuration)
+- [jq](https://stedolan.github.io/jq/)
+
+Many of the following commands in this guide, as well as the commands [operations](./operations.md), use environment variables. Export the appropriate values for the upgrade so you don't lose track:
+
+```sh
+# name of customer deployment (should match folder)
+export CUSTOMER=<customer_or_instance_name>
+
+# the previous actvie deployment, either `red` or `black`
+# this can be determined by obtaining the last snapshot name by running `gcloud compute snapshots list`
+# `default-red-data-disk-snapshot--upgrade-from-<>` -> `red`
+# `default-black-data-disk-snapshot--upgrade-from-<>` -> `black`
+export CURRENT_DEPLOYMENT="[red|black]"
+```
+
+Make sure your copy of the [deploy-sourcegraph-managed]() repository is up to date:
+
+```sh
+git checkout main
+git pull origin main
+```
+
+## Sourcegraph resume
+
+### 0) Sourcegraph resume setup
+
+**Follow the [general setup](#general-setup) guide.** Then, set up the appropriate version variables:
+
+Make sure to use the same shell for all the commands in this guide unless otherwise stated.
+
+Now start a branch for your upgrade:
+
+```sh
+git checkout -b $CUSTOMER/resume-instance
+# all the below steps are documented assuming you are in the customer deployment directory
+cd $CUSTOMER
+```
+
+### 1) Prepare terraform module
+
+Uncomment out the following resources in `infrastructure.tf`.
+
+- `google_compute_network_endpoint.primary`
+- `google_monitoring_uptime_check_config.primary`
+- `google_monitoring_alert_policy.primary`
+
+Update `terraform.tfvars`, replace `<snapshot_name>` with the snapshot taken during suspend process. Depending on the previous active deployment, the disk to restore from may vary. Run `gcloud compute snapshots list` to obtain the snapshot name.
+
+If the `$CURRENT_DEPLOYMENT` is `red`, vice versa.
+
+```tf
+disks = {
+  red = { from_snapshot = "default-red-data-disk--upgrade-from-<>" }
+}
+```
+
+```sh
+terraform plan -out resume.plan
+```
+
+If everything look good to you, apply the plan
+
+```sh
+terraform apply resume.plan
+```
+
+### 2) Make database writable
+
+You should ssh into the deployment and wait until postgres services are healthy ([instructions](operations.md#ssh-access)), then run the following.
+
+```sh
+../util/set-db-readonly.sh $CURRENT_DEPLOYMENT false
+```
+
+### 3) Verify instance heath
```

Comment:
We're missing a step here - we should also check that the instance is running the latest version of Sourcegraph, and perform any upgrades necessary.